Sources/NostrSDK/Events/Tags/EventCoordinates.swift

```diff
@@ -85,13 +85,49 @@ public struct EventCoordinates: PubkeyProviding, RelayProviding, RelayURLValidat
         self.tag = eventCoordinatesTag
     }
 
-    /// Initializes coordinates to a replaceable event.
+    /// Initializes coordinates to a non-parameterized replaceable event.
+    /// Returns nil if the kind is not a non-parameterized replaceable event kind.
     /// - Parameters:
-    ///   - kind: The ``EventKind`` of the referenced replaceable event.
-    ///   - pubkey: The pubkey that signed the referenced replaceable event.
-    ///   - identifier: The identifier of the referenced replaceable event.
-    ///   - relayURL: A relay in which the referenced replaceable event could be found.
-    public init?(kind: EventKind, pubkey: PublicKey, identifier: String?, relayURL: URL? = nil) {
+    ///   - kind: The ``EventKind`` of the referenced non-parameterized replaceable event.
+    ///   - pubkey: The pubkey that signed the referenced non-parameterized replaceable event.
+    ///   - relayURL: A relay in which the referenced non-parameterized replaceable event could be found.
+    public init?(kind: EventKind, pubkey: PublicKey, relayURL: URL? = nil) {
```

Comment:
Couldn't we have `identifier` still here and default it to empty string if not provided? Then this initializer could be used for either kind? You could still validate the "parameterized or not" part based on the kind. I might be missing something here.

Also I think it would be confusing to only return nil here. IMO we should either put an assert here with a relevant message or have this initializer throw an error with a human-readable message.